### DISCUSSION TITLE
CODICE PER SALUTO BENVENUTO A CASA

### DISCUSSION THREAD

**Discussion by elsapopping** [indomus/forum#2446] [HUB: Home Assistant]
salve a tutti, vorrei sapere se è possibile scrivere un codice che mi permetta di compiere un'azione quando una persona è presente in casa.
in particolare vorrei accogliere coloro che entrano in casa. Non mi serve il codice di verifica della presenza, ma la routine con cui eseguire il saluto. ad esempio se entrate tutti i componenti della famiglia salutate tutti, altrimenti salutate solo chi è entrato

**Comment by matteo-vannucchi** [comment]
Ciao @elsapopping , puoi iniziare da [questo articolo](https://indomus.it/formazione/uscire-e-rientrare-a-casa-lautomazione-domotica-via-home-assistant/) su inDomus, anch'io voglio ottenere una cosa simile a quella che hai descritto e sono partito da lì

**Reply by elsapopping** [reply]
Ciao e grazie per il consiglio. Il problema che non riesco a risolvere non riguarda la situazione  in cui tutti i membri della famiglia ( raggruppati in "group") rientrano insieme, ma le possibili varianti, ad esempio due dei quattro o tre dei quattro. Come faccio a creare una situazione che deve escludere tutte le altre? Mi spiego meglio se invece di rientrare A B C D, rientrano solo A B, quale istruzioni utilizzo?

**Reply by Scialla** [reply]
non devi pensare a escludere chi non c'è ma a includere chi c'è; utilizza come `trigger` non il gruppo ma le singole `person` (quindi un `trigger` per ogni persona poi, nella `action` usi il `trigger.entity_id` per salutare la persona che ha fatto triggerare l'automazione ed usi come `mode: queued` in modo che se entrano più persone insieme l'automazione triggeri più volte accodando i messaggi
